File: cases/python_openai_hosted_mcp_approval/star_agent.py
from agents import Agent, HostedMCPTool
from star_barrel import (
    IMPORTED_ALWAYS as STAR_REEXPORTED_ALWAYS,
    IMPORTED_TOOL_CONFIG as STAR_REEXPORTED_TOOL_CONFIG,
    MUTATED_TOOL_CONFIG as STAR_REEXPORTED_MUTATED_TOOL_CONFIG,
)


def build_star_agent() -> Agent:
    return Agent(
        name="Python hosted MCP star reexport approval agent",
        tools=[
            HostedMCPTool(
                tool_config={
                    "type": "mcp",
                    "server_label": "star_reexported_literal",
                    "server_url": "https://mcp.example.com/mcp",
                    "require_approval": STAR_REEXPORTED_ALWAYS,
                }
            ),
            HostedMCPTool(tool_config=STAR_REEXPORTED_TOOL_CONFIG),
            HostedMCPTool(tool_config=STAR_REEXPORTED_MUTATED_TOOL_CONFIG),
        ],
    )

File: cases/typescript_openai_hosted_mcp_approval/star-agent.ts
import { Agent, hostedMcpTool } from "@openai/agents";
import { importedPolicy as starPolicy } from "./star";
import { importedPolicy as ambiguousStarPolicy } from "./ambiguous-star";

const starAgent = new Agent({
  name: "Hosted MCP star policy agent",
  tools: [
    hostedMcpTool({
      serverLabel: "star",
      serverUrl: "https://mcp.example.com/mcp",
      requireApproval: starPolicy,
    }),
    hostedMcpTool({
      serverLabel: "ambiguous-star",
      serverUrl: "https://mcp.example.com/mcp",
      requireApproval: ambiguousStarPolicy,
    }),
  ],
});
void starAgent;
